### PR TITLE
ExpandJSON transformer

### PR DESCRIPTION
**Pull request by TamerShlash**
This transformer parses values of specified files as JSON and expands the resulting sub-fields to the original record.
See the test cases to learn more.

### CODE REVIEW

File: lib/optimus_prime/transformers/expand_json.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+      def write(record)
+        transformed_record = transform(record)
+        push transformed_record if transformed_record
+      end
+
+      private
+
+      def transform(record)
+        result = record.deep_dup
```
Comment (omarkhan) [inline lib/optimus_prime/transformers/expand_json.rb]: A shallow copy should be enough here right?

Comment (TamerShlash) [inline lib/optimus_prime/transformers/expand_json.rb]: Yes it should be enough.

Comment (omarkhan) [inline lib/optimus_prime/transformers/expand_json.rb]: Then let's drop the `deep_dup` marshal hack and just use `dup` here

Comment (TamerShlash) [inline lib/optimus_prime/transformers/expand_json.rb]: @omarkhan :+1:

**Comment by Anumodana** [comment]
:+1:

**Comment by rickapichairuk** [comment]
Good stuff Tamer!

**Pull request merged**